### PR TITLE
Docstyle formatting

### PR DESCRIPTION
**Pull request by NKeleher**
Enabling the docstyle linting and formatting with Ruff rule "D".

Changes:
- Uncommented the "D" linting rule
- Set docstyle line width to 88
- standardized docstrings to the numpy docstring format style: https://numpydoc.readthedocs.io/en/latest/format.html#sections
- minor spelling fixes

I've tried to standardize the format and conventions as much as possible.

I ran quartodoc to confirm that docs build. I also ran tests to confirm that they pass.

### CODE REVIEW

**Comment by NKeleher** [comment]
The should be the last big PR to implement ruff lintingband formatting. 🤞

File: pyfixest/FormulaParser.py [hunk 1]
```diff
@@ -47,15 +54,15 @@ def __init__(self, fml):
             if "~" in fml_split[1]:
                 fevars = "0"
                 endogvars, instruments = fml_split[1].split("~")
-                # add endogeneous variable to "covars" - yes, bad naming
+                # add endogenous variable to "covars" - yes, bad naming
```
Comment (s3alfisc) [inline pyfixest/FormulaParser.py:57]: Almost 10 years of econometrics and I still haven't learned how to write endogenous correctly ... 😅

File: pyfixest/did/lpdid.py [hunk 2]
```diff
@@ -9,6 +9,37 @@
 
 
 class LPDID(DID):
+    """
+    A class used to represent the Local Polynomial Differences-in-Differences model.
```
Comment (s3alfisc) [inline pyfixest/did/lpdid.py:13]: Here it should say the "local projections"  estimator. Took me a while, I'd always say "linear projections" instead ... 😄

File: pyfixest/utils.py [hunk 3]
```diff
@@ -9,14 +9,18 @@ def ssc(adj=True, fixef_k="none", cluster_adj=True, cluster_df="min"):
...
-            and k is the number of estimated coefficients excluding any fixed effects projected out in either fixest::feols() or lfe::felm().
-        fixef_k: str, default "none"
-            Equal to 'none': the fixed effects parameters are discarded when calculating k in (N-1) / (N-k).
-        cluster_adj: bool, default True
-            If True, a cluster correction G/(G-1) is performed, with G the number of clusters.
-        cluster_df: str, default "conventional"
+        adj : bool, default True
+            If True, applies a small sample correction of (N-1) / (N-k) where N
+            is the number of observations and k is the number of estimated
+            coefficients excluding any fixed effects projected out in either
```
Comment (s3alfisc) [inline pyfixest/utils.py:15]: Here I seem to have copied the docs of one of my R packages, `fwildclusterboot`...

**Review by s3alfisc — APPROVED** [review]

**Comment by s3alfisc** [comment]
Perfect, it's merged! Thanks so much @NKeleher!

**Pull request merged**